### PR TITLE
Scripts: Convert template script engines to return String instead of BytesReference

### PR DESCRIPTION
**Pull request by rjernst**
Template script engines (mustache, the only one) currently return a
BytesReference that users must know is utf8 encoded. This commit
modifies all callers and mustache to have the template engine return
String. This is much simpler, and does not require decoding in order to
use (for example, in ingest).

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

File: modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustacheScriptEngineService.java [hunk 1]
```diff
@@ -58,21 +59,6 @@
 
     public static final String NAME = "mustache";
 
-    /** Thread local UTF8StreamWriter to store template execution results in, thread local to save object creation.*/
-    private static ThreadLocal<SoftReference<UTF8StreamWriter>> utf8StreamWriter = new ThreadLocal<>();
```
Comment (martijnvg) [inline modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustacheScriptEngineService.java]: :+1

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
@elasticmachine retest this please

**Comment by rjernst** [comment]
@elasticmachine retest this please

**Pull request merged**